content/technotes/2020-07-09-cran-checks-update/index.md

```diff
@@ -0,0 +1,196 @@
+The email is structured as follows:
+
+{{<figure src="email.png" alt="An example email users get from the CRAN checks API notifications service, including the rule triggered and link to check results." width="450">}}
+
+- Triggered rule: with a list of each of the four categories and their values.
+- Date of the check result (matches closely the date on the CRAN check results that CRAN maintainers created)
+- Your check results: link to the CRAN checks API JSON output for your package
+- The rest is information, report bugs, ask for help, etc. Note the "Unsubscribe" link doesn't actually work yet!
+
+#### Managing notifications
+
+There is no web interface to notifications at this time. The only official interface is the cchecks R package - you are welcome to interact with the API itself via curl or any other tool.
+
+First, you'll need to register for a token (aka key). Using [cchn_register()](https://docs.ropensci.org/cchecks/reference/cchn_register), you run the function with or without an email address. If no email address is given we look for an email address in [various places](https://github.com/ropenscilabs/cchecks/blob/53b70365546e617e70f602a9bae8b65207e730b3/R/email-utils.R#L81-L129), and ask you which one you'd like to use, or you can supply one at the prompt.
+
+```r
+cchn_register()
+```
+
+Running `cchn_register()` caches the token in a file locally on your computer. 
+
+After registering, you can manage rules for your packages. There's two ways to manage rules: a) across packages, or b) in a single package context.
+
+Functions prefixed with `cchn_pkg_` operate within a package directory. That is, your current working directory is an R package, and is the package for which you want to handle CRAN checks notifications. These functions make sure that you are inside of an R package, and use the email address and package name based on the directory you're in.
+
+Functions prefixed with just `cchn_` do not operate within a package. These functions do not guess package name at all, but require the user to supply a package name (for those functions that require a package name); and instead of guessing an email address from your package, we guess email from the cached email/token file.
+
+If you don't use the package specific functions, you can add a rule like:
+
+```r
+cchn_rule_add(package = "foobar", status = "warn", platform = 2)
+```
+
+Using package specific functions, you can add a rule like:
+
+```r
+cchn_pkg_rule_add(status = "warn", platform = 2)
+```
+
+See `cchn_rules_add()` for adding many rules at once.
+
+What the first author does currently is a rule for each of his packages checking for a status of **ERROR** for at least **2 days**. You could take this approach as well for your packages. We are thinking about ways to build in some sensible default rules, as well as making it easier to work across all of your packages by [looking them up for you by your maintainer email](https://github.com/ropenscilabs/cchecks/issues/23).
```

Comment:
could you re-add the function call corresponding to this rule, or just note that it corresponds to the two examples above? :slightly_smiling_face:

Reply:
added example of how I use `cchn_rules_add`